bindings/rust/bench/src/openssl.rs

```diff
@@ -0,0 +1,119 @@
+    fn handshake_conn(&mut self, mode: Mode) -> Result<(), Box<dyn Error>> {
+        match match mode {
+            Mode::Client => self.client_conn.connect(),
+            Mode::Server => self.server_conn.accept(),
+        } {
+            Ok(_) => Ok(()),
+            Err(err) => {
+                if err.code() != ErrorCode::WANT_READ {
+                    Err(err.into())
+                } else {
+                    Ok(())
+                }
+            }
+        }
+    }
+}
+
+impl TlsBenchHarness for OpenSslHarness {
+    fn new(crypto_config: &CryptoConfig) -> Result<Self, Box<dyn Error>> {
+        let client_buf = ConnectedBuffer::new();
+        let server_buf = client_buf.clone_inverse();
+
+        let cipher_suite = match crypto_config.cipher_suite {
+            CipherSuite::AES_128_GCM_SHA256 => "TLS_AES_128_GCM_SHA256",
+            CipherSuite::AES_256_GCM_SHA384 => "TLS_AES_256_GCM_SHA384",
+        };
+
+        let ec_key = match crypto_config.ec_group {
+            ECGroup::SECP256R1 => "P-256",
+            ECGroup::X25519 => "X25519",
+        };
+
+        let mut client_builder = SslConnector::builder(SslMethod::tls())?;
+        client_builder.set_ca_file(CA_CERT_PATH)?;
+        Self::common_config(&mut client_builder, cipher_suite, ec_key)?;
+
+        // SslAcceptorBuilder has to have set of safe defaults
+        // Arbitrarily chose mozilla_modern_v5(), override defaults
+        let mut server_builder = SslAcceptor::mozilla_modern_v5(SslMethod::tls())?;
+        server_builder.set_certificate_chain_file(SERVER_CERT_CHAIN_PATH)?;
+        server_builder.set_private_key_file(SERVER_KEY_PATH, SslFiletype::PEM)?;
+        Self::common_config(&mut server_builder, cipher_suite, ec_key)?;
+
+        let client_config = client_builder.build().into_context();
+        let server_config = server_builder.build().into_context();
+
+        let client_conn = SslStream::new(Ssl::new(&client_config)?, client_buf)?;
+        let server_conn = SslStream::new(Ssl::new(&server_config)?, server_buf)?;
+
+        Ok(Self {
+            client_conn,
+            server_conn,
+        })
+    }
+
+    fn handshake(&mut self) -> Result<(), Box<dyn Error>> {
+        for _ in 0..2 {
+            self.handshake_conn(Mode::Client)?;
+            self.handshake_conn(Mode::Server)?;
+        }
+        Ok(())
+    }
+
+    fn handshake_completed(&self) -> bool {
+        self.client_conn.ssl().is_init_finished() && self.server_conn.ssl().is_init_finished()
+    }
+
+    fn get_negotiated_cipher_suite(&self) -> CipherSuite {
+        match self
+            .client_conn
+            .ssl()
+            .current_cipher()
+            .expect("Handshake not completed")
+            .name()
+        {
+            "TLS_AES_128_GCM_SHA256" => CipherSuite::AES_128_GCM_SHA256,
+            "TLS_AES_256_GCM_SHA384" => CipherSuite::AES_256_GCM_SHA384,
+            _ => panic!("Unknown cipher suite"),
+        }
+    }
```

Comment:
Can we break this down with an intermediate variable? The change on match makes it a little bit higher friction to reason through what is actually being matched one.
```suggestion
    fn get_negotiated_cipher_suite(&self) -> CipherSuite {
        let cipher_suite = self
            .client_conn
            .ssl()
            .current_cipher()
            .expect("Handshake not completed")
            .name();
        match cipher_suit {
            "TLS_AES_128_GCM_SHA256" => CipherSuite::AES_128_GCM_SHA256,
            "TLS_AES_256_GCM_SHA384" => CipherSuite::AES_256_GCM_SHA384,
            _ => panic!("Unknown cipher suite"),
        }
    }
```